README_cppyy.md

```diff
@@ -0,0 +1,187 @@
+Cppyy is a library for automatically generating python bindings for C++ code. This allows us to call any function from NESO in python code.
+As an example, the whole of NESO main is replicated in `cppyy_run.py`.
+
+Cppyy requires llvm / clang to build. I've had problems setting up the build
+environment automatically, but this set up works for me.
+
+## Activate the spack environment
+
+```
+git submodule update --init
+. activate
+```
+
+## Set up the Python environment
+
+Install the python dependencies in a virtual environment by doing
+
+```
+python3 -m venv venv_cppyy
+. venv_cppyy/bin/activate
+pip3 install -r requirements
+```
+
+This will install the dependencies:
+
+* cppyy==2.4.2
+* libclang
+* clang-format
+* setuptools
+* wheel
+
+# Install the spack environment
+
+Set up the `spack.yaml` to the supported toolchain:
+
+```
+spack:
+  specs:
+  - neso%clang@12.0.1 ^openblas ^hipsycl ^scotch@6 ^llvm@12.0.1 ^python@3.9.12%gcc@11.2.0
```

Comment:
In the end we'll want to have a spec per build, so something like
```
- neso%clang@12.0.1 ^openblas ^hipsycl@0.9.2%gcc@11.2.0 ^scotch@6 ^llvm@12.0.1%gcc@11.2.0  ^python@3.9.12%gcc@11.2.0 ^clang
```

Preferably with a set of allowable clangs and gccs.